realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -543,8 +543,8 @@ public void setObject(String fieldName, DynamicRealmObject value) {
             if (value.realm == null || value.row == null) {
                 throw new IllegalArgumentException("Cannot link to objects that are not part of the Realm.");
             }
-            if (!realm.getConfiguration().equals(value.realm.getConfiguration())) {
-                throw new IllegalArgumentException("Cannot add an object from another Realm");
+            if (realm != value.realm) {
+                throw new IllegalArgumentException("Cannot add an object from another Realm.");
```

Comment:
Different Realm is a bit ambiguous here? Because this also fails if you have to SharedGroups for the same Realm. I guess that is the point though, but perhaps rephrase to `Cannot add an object from another Realm instance` ?